backend/core/src/main/java/org/sonarsource/sonarlint/core/ConnectionSuggestionProvider.java

```diff
@@ -0,0 +1,138 @@
+@Named
+@Singleton
+public class ConnectionSuggestionProvider {
+
+  private static final SonarLintLogger LOG = SonarLintLogger.get();
+
+  private final ConfigurationRepository configRepository;
+  private final ConnectionConfigurationRepository connectionRepository;
+  private final SonarLintRpcClient client;
+  private final BindingClueProvider bindingClueProvider;
+  private final ExecutorServiceShutdownWatchable<?> executorService;
+  private final BindingSuggestionProvider bindingSuggestionProvider;
+
+  @Inject
+  public ConnectionSuggestionProvider(ConfigurationRepository configRepository, ConnectionConfigurationRepository connectionRepository, SonarLintRpcClient client,
+                                   BindingClueProvider bindingClueProvider, BindingSuggestionProvider bindingSuggestionProvider) {
+    this.configRepository = configRepository;
+    this.connectionRepository = connectionRepository;
+    this.client = client;
+    this.bindingClueProvider = bindingClueProvider;
+    this.executorService = new ExecutorServiceShutdownWatchable<>(new ThreadPoolExecutor(0, 1, 10L, TimeUnit.SECONDS,
+      new LinkedBlockingQueue<>(), r -> new Thread(r, "Connection Suggestion Provider")));
+    this.bindingSuggestionProvider = bindingSuggestionProvider;
+  }
+
+  @EventListener
+  public void filesystemUpdated(FileSystemUpdatedEvent event) {
+    var listConfigScopeIds = event.getAddedOrUpdated().stream()
+      .filter(f -> ALL_BINDING_CLUE_FILENAMES.contains(f.getFileName()) || f.isSonarlintConfigurationFile())
+      .map(ClientFile::getConfigScopeId)
+      .collect(Collectors.toSet());
+
+    if (!listConfigScopeIds.isEmpty()) {
+      var cancelMonitor = new SonarLintCancelMonitor();
+      cancelMonitor.watchForShutdown(executorService);
+      executorService.submit(() -> suggestConnectionForGivenScopes(listConfigScopeIds, cancelMonitor));
+    }
+  }
+
+  private void suggestConnectionForGivenScopes(Set<String> listOfFilesPerConfigScopeIds, SonarLintCancelMonitor cancelMonitor) {
+    LOG.debug("Computing connection suggestions");
+    var connectionSuggestionsByConfigScopeIds = new HashMap<String, ConnectionSuggestionDto>();
+    for (var configScopeId : listOfFilesPerConfigScopeIds) {
+      var effectiveBinding = configRepository.getEffectiveBinding(configScopeId);
+      if (effectiveBinding.isPresent()) {
+        LOG.debug("A binding already exist, skipping the connection suggestion");
+        continue;
+      }
+
+      var bindingClues = bindingClueProvider.collectBindingClues(configScopeId, cancelMonitor);
+      if (bindingClues.size() == 1) {
+        var bindingClue = bindingClues.get(0);
+        var projectKey = bindingClue.getSonarProjectKey();
+        if (projectKey != null) {
+          handleBindingClue(connectionSuggestionsByConfigScopeIds, bindingClue, configScopeId, projectKey);
+        }
+      } else if (bindingClues.size() > 1) {
+        LOG.debug("Multiple binding clues found, skipping the suggestion");
```

Comment:
I think we could try to better handle this case: we could suggest several connections. Or suggest one of them. I agree this needs to be adapted in the protocol and the UI

Reply:
After discussing with the sprint members, we'll send all the suggestions to the client, and it will be up to him to decide how to choose